### DISCUSSION TITLE
Help: Can not delete photos

### DISCUSSION THREAD

**Discussion by MrCoala** [photoprism/photoprism#885] [Q&A] [answered]
Hello,

i started to import my whole photo library but it seems that i can not delete any pictures. Where can i find that option?

**Comment by lastzero** [comment]
We need your financial support to develop it. See public roadmap. Thank you very much! ❤️

**Reply by alexisLefebvre** [reply]
Was it removed recently? I think it was possible some months ago to remove photos from the UI.

**Reply by graciousgrey** [reply]
No it was never possible to delete photos using the UI. 
The only thing you can delete in the UI are not primary files of stacks. 
All other files need to be deleted in the originals folder directly until https://github.com/photoprism/photoprism/issues/167 is implemented.

**Reply by tootbrute** [reply]
> Was it removed recently? I think it was possible some months ago to remove photos from the UI.

You can "remove photos" from the UI by Archiving them. Then, in the future, when deletion is supported, you may easily delete them. This is what I'm doing at the moment.

**Reply by alexisLefebvre** [reply]
> No it was never possible to delete photos using the UI.
> The only thing you can delete in the UI are not primary files of stacks.
> All other files need to be deleted in the originals folder directly until #167 is implemented.

Actually it's possible to remove a video:

> ![image](https://user-images.githubusercontent.com/2071331/104735950-8a9dfd80-5742-11eb-92d6-440adae93b95.png)

https://demo.photoprism.org/browse

**Reply by lastzero** [reply]
As she wrote: "The only thing you can delete in the UI are not primary files of stacks."

The JPEG above is the primary file.

**Reply by alexisLefebvre** [reply]
Oh, that's right. Thanks.

**Comment by tootbrute** [comment]
> Hello,
> 
> i started to import my whole photo library but it seems that i can not delete any pictures. Where can i find that option?

You can delete photos manually, then reindex in Photoprism.

But it's true, I encourage everybody using this should kick in a few dollars to help them develop it more :+1:

**Comment by graciousgrey** [comment, marked as answer]
The delete functionality is now available for our sponsors as early access feature :)

**Reply by gsownsby** [reply]
Hmm...with all due respect, withholding delete function (batch or simplistic multi-file deletion) is a bit like giving a car buyer a test drive with no brakes.

**Reply by lastzero** [reply]
With all due respect, just enable it in Settings.

**Reply by gsownsby** [reply]
Thank you for your reply.  'Sorry for my venting and frustration in my post; however, having only uploaded less than 35 images which spawned the generation of 17,764 images by PhotoPrism, I have buckets of the same photos or pieces of those photos with no apparent easy way to do a batch deletion.  'Still trying to get an understanding of why PhotoPrism did what it did.  If A.I. spawned all of that, then that would become a real problem for me with 100,000+ images...just trying to understand what I'm seeing.

**Reply by graciousgrey** [reply]
This sounds like your storage folder is located insides of your originals folder.

>never configure the storage folder to be inside the originals folder unless the name starts with a . to indicate that it is  hidden

https://docs.photoprism.app/getting-started/docker-compose/#volumes

You can multi select photos as described [here](https://docs.photoprism.app/user-guide/navigate/#selection-mode-and-multi-select), move them to the archive and [delete](https://docs.photoprism.app/user-guide/organize/delete/) them.

Make sure to fix your configuration before you trigger the next indexing.

**Reply by gsownsby** [reply]
Thank you for your reply.  I'd be happy to check the location though I don't think I did it that way but I will check.

If all else fails, how can I flush everything and start anew?

Thanks!

**Reply by gsownsby** [reply]
> This sounds like your storage folder is located insides of your originals folder.
> 
> > never configure the storage folder to be inside the originals folder unless the name starts with a . to indicate that it is  hidden
> 
> https://docs.photoprism.app/getting-started/docker-compose/#volumes
> 
> You can multi select photos as described [here](https://docs.photoprism.app/user-guide/navigate/#selection-mode-and-multi-select), move them to the archive and [delete](https://docs.photoprism.app/user-guide/organize/delete/) them.
> 
> Make sure to fix your configuration before you trigger the next indexing.

I'm happy to report that I did error in my configuration of the folders.  I completely restarted from scratch and all is well now.  Thanks again.

**Comment by qrhfz** [comment]
is delete free in 2024? if so, how to enable it?

**Reply by lastzero** [reply]
It's been free for a long time. See our User Guide for how to enable and use it! :)
